### ISSUE TITLE
Aperture photometry with a different aperture for each object

### ISSUE THREAD

**Issue by dlakaplan** [astropy/photutils#748]
`CircularAperture` / `SkyCircularAperture` appear to require that the radius be a single value for all objects.  This is not always desirable.  For instance, for forced photometry on extended objects one wishes to match the prior apertures.  Instead one now has to create individual apertures and iterate over them.

**Comment by larrybradley** [comment]
@dlakaplan This feature was removed from `photutils` aperture objects many years ago (it was there in the very beginning).  There was a lengthy discussion among the developers 4 years ago about adding it back (https://github.com/astropy/photutils/issues/62 and related issues), but no clear consensus.

It's not as straightforward as simply allowing the circular radius parameter to be a list/array of values.  To some people this means that radius should be broadcast to all positions and to others it means position[i] at radius[i].  Also, it is even more complicated to generalize to apertures defined by multiple parameters, i.e. what would it mean for `EllipticalAnnlus` to have:

```python
EllipticalAnnulus([pos0...posN], a_in=[a0...aN], a_out=[o0...oN], b_out=[b0...bN], theta=[t0...tN])
```
For a consistent broadcasting API, the above would produce a photometry table with `aN * oN * bN * tN` columns, and good luck parsing the output table to figure out what shape parameters were use for each.  Ugh!

An API allowing the shape parameters to be iterables is so ambiguous (and contentious) that I don't think it makes much sense to allow it.  (btw, the `regions` package goes one step further where a region object refers to a single *position* and *shape*).

The current approach in `photutils` is that an aperture instance refers to an object with a single *shape* (but it can have multiple positions).  That makes the API very clear and straightforward.

The broadcasting use case of multiple aperture shapes at each input position is very easy.   You do need to create separate aperture objects of different shapes, but you simply need to pass a list of these apertures (easily made with a list comprehension) to `aperture_photometry` and it will do the iterations, giving you a single output table (see https://photutils.readthedocs.io/en/stable/aperture.html#multiple-apertures-at-each-position).  And there's no ambiguity in the output table either.

If you are doing something more complicated with multiple aperture shapes, also note that you can directly use the `do_photometry` aperture method.

**Comment by dlakaplan** [comment]
I appreciate the information.  I searched open issues for discussions of this, but didn't look through the whole history.  I implemented this for some subset of use cases (and suggested a pull request) but I guess that it can indeed be much more complicated.  For whatever reason I didn't find the doc page that you suggest, which would indeed work in my case.  I tried to loop over all apertures individually but that was very slow (separate overhead for each image) and made the output complicated.
